hnn_core/io.py

```diff
@@ -0,0 +1,394 @@
+    net_data['pos_dict'] = pos_dict_data
+    # Write cell_response
+    if (not net.cell_response) or save_unsimulated:
+        net_data['cell_response'] = None
+    else:
+        net_data['cell_response'] = net.cell_response.to_dict()
+    # Write External drives
+    external_drives_data = dict()
+    for key in net.external_drives.keys():
+        external_drives_data[key] = (_external_drive_to_dict
+                                     (net.external_drives[key],
+                                      save_unsimulated))
+    net_data['external_drives'] = external_drives_data
+    # Write External biases
+    net_data['external_biases'] = net.external_biases
+    # Write connectivity
+    net_data['connectivity'] = _write_connectivity(net.connectivity)
+    # Write rec arrays
+    net_data['rec_arrays'] = dict()
+    for key in net.rec_arrays.keys():
+        rec_array_copy = net.rec_arrays[key].copy()
+        if save_unsimulated:
+            rec_array_copy._reset()
+        net_data['rec_arrays'][key] = rec_array_copy.to_dict()
+    # Write threshold
+    net_data['threshold'] = net.threshold
+    # Write delay
+    net_data['delay'] = net.delay
+
+    # Saving file
+    write_hdf5(fname, net_data, overwrite=overwrite)
+
+
+@fill_doc
+def read_network(fname, read_raw=False):
+    """Read network from a file.
+
+    Parameters
+    ----------
+    %(fname)s
+    %(read_raw)s
+
+    Outputs
+    -------
+    %(net)s
+
+    File Content Description
+    ------------------------
+    %(network_file_content_description)s
+
+    cell type description
+    ---------------------
+    %(cell_description)s
+
+    gid range description
+    ---------------------
+    %(gid_range_description)s
+
+    external drive description
+    --------------------------
+    %(external_drive_description)s
+
+    external bias description
+    -------------------------
+    %(external_bias_description)s
+
+    connection description
+    ----------------------
+    %(connection_description)s
+
+    extracellular array description
+    -------------------------------
+    %(extracellular_array_description)s
+    """
+    net_data = read_hdf5(fname)
+    if 'object_type' not in net_data:
+        raise NameError('The given file is not compatible. '
+                        'The file should contain information'
+                        ' about object type to be read.')
+    if net_data['object_type'] != 'Network':
+        raise ValueError('The object should be of type Network. '
+                         'The file contains object of '
+                         'type %s' % (net_data['object_type'],))
+    params = dict()
+    params['N_pyr_x'] = net_data['N_pyr_x']
+    params['N_pyr_y'] = net_data['N_pyr_y']
+    params['celsius'] = net_data['celsius']
+    params['threshold'] = net_data['threshold']
+
+    # Instantiating network
+    # Cannot do this globally due to circular import
+    from .network import Network
+    net = Network(params)
```

Comment:
put the import at the top of the function

Reply:
I would actually do the nested import in the `write` method

Reply:
We are using `copy doc` decorator for `write` method, hence importing `write_network` inside `net.write` is not working. Doing it inside the class leads to circular import.